### PR TITLE
Move all sorting/filtering logic into MainPage component

### PR DESCRIPTION
**Pull request by sunify**
- In App only sort/filter params stored
- No data duplication because of cache
- Fixed strange setState statements
- Implemented sorting

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -781,54 +771,36 @@ export default class App extends Component {
       voteEndTime,
       voteStartTime
     } = body.contents;
-    const proposals = proposalsList.filter(p => p.proposalId)
-    this.setState(state => ({
-      ...state,
+    const proposals = proposalsList.filter(p => p.proposalId).filter((p, i, list) => list.findIndex(p2 => p2.proposalId === p.proposalId) === i);
```
Comment (troggy) [inline src/App.js:775]: what is the second `filter` call for?

Comment (sunify) [reply src/App.js:775]: It's temporary, but now we have couple of proposal with same id in store. Probably some comment needed here

File: src/MainPage.js [hunk 2]
```diff
@@ -4,44 +4,68 @@ import ProposalsList from "./volt/components/ProposalsList";
...
+  sortingOrder,
+  toggleFavorites,
+  filterQuery,
+  favorites,
+  voteStartTime,
+  voteEndTime,
+}) {
+  const filteredList = React.useMemo(() => {
+    const sortFunctions = {
+      votes: (a, b) => 0,
```
Comment (troggy) [inline src/MainPage.js:24]: not implemented?

Comment (sunify) [reply src/MainPage.js:24]: AFAIK we don't have this data yet

**Review by troggy — APPROVED** [review]
Code looks and works good 👍

**Review by troggy — APPROVED** [review]

**Review by johba37 — APPROVED** [review]

**Pull request merged**